torch/distributed/launch.py

```diff
@@ -203,6 +207,9 @@ def main():
     current_env["MASTER_PORT"] = str(args.master_port)
     current_env["WORLD_SIZE"] = str(dist_world_size)
 
+    if 'OMP_NUM_THREADS' not in os.environ:
+        current_env["OMP_NUM_THREADS"] = str(int(multiprocessing.cpu_count() / args.nproc_per_node))
```

Comment:
There would still be waste if `multiprocessing.cpu_count() / args.nproc_per_node` is not an integer. I would move the setting of `OMP_NUM_THREAD` down into the loop, and have the number of threads be `multiprocessing.cpu_count() / args.nproc_per_node + 1` if `local_rank < multiprocessing.cpu_count() % args.nproc_per_node`.

Reply:
The optimal number of OMP threads might not simply be nb of CPU / nb of processes per node, because there is also data loader that should be taken into account.

See https://github.com/pytorch/vision/pull/818#issuecomment-508465650 for example.

This is something that we should raise awareness of in some way, or else people will have much slower training times for some models and they won't know why.

Reply:
Very good point. We could choose to log the default value, just to call attention that the value is set in the first place. Then, if people have performance issues, there is at least a hint about where to look.